text/3537-msrv-resolver.md

```diff
@@ -0,0 +1,1238 @@
+  - Needed for [verifying latest dependencies](https://doc.rust-lang.org/nightly/cargo/guide/continuous-integration.html#verifying-latest-dependencies)
+- `minimum` (unstable): `-Zminimal-versions`
+  - As this is just precedence, `-Zdirect-minimal-versions` doesn't fit into this
+- `rust-version`:  what is defined in the package (default for v3 resolver)
+- `rust-version=` (future possibility)
+  - `package`: long form of `rust-version`
+  - `rustc`: the current running version
+    - Needed for "separate development / publish MSRV" workflow
+  - `<x>[.<y>[.<z>]]`: manually override the version used
+
+If a `rust-version` value is used, we'd switch to `maximum` when `--ignore-rust-version` is set.
+This will let users effectively pass `--ignore-rust-version` to all commands,
+without having to support the flag on every single command.
+
+## `cargo build`
+
+The MSRV-compatibility build check will be demoted from an error to a `deny`-by-default workspace
+[diagnostic](https://github.com/rust-lang/cargo/issues/12235),
+allowing users to intentionally use dependencies on an unsupported (or less supported) version of Rust
+without requiring `--ignore-rust-version` on every invocation.
+
+Ideally, we present all of the MSRV issues upfront to be resolved together.
+At minimum, we should present a top-down message, rather than bottom up.
+
+If `package.rust-version` is unset or `"auto"`, the diagnostic should suggest setting it
+to help raise awareness of `package.rust-version` being able to reduce future
+resolution errors.
+This would benefit from knowing the oldest MSRV.
+
+## `cargo update`
+
+`cargo update` will inform users when an MSRV or semver incompatible version is available.
+`cargo update --dry-run` will also report this information so that users can check on the status of this at any time.
+
+Users may pass
+- `--ignore-rust-version` to pick the latest dependencies, ignoring all `rust-version` fields (your own and from dependencies)
+- `--update-rust-version` to pick the `rustc --version`-compatible dependencies, updating your `package.rust-version` if needed to match the highest of your dependencies
+- `<pkgname> --precise <version>` to pick a specific version, independent of the `rust-version` field
+
+We expect the notice to inform users of these options for allowing them to upgrade.
+
+Those flags will also be added to `cargo generate-lockfile`
+
+## Syncing `Cargo.toml` to `Cargo.lock` on any Cargo command
+
+In addition to the `cargo update` output to report when things are held back (both MSRV and semver),
+we will try having dependency resolves highlight newly selected dependency versions that were held back due to MSRV or semver.
+Whether we do this and how much will be subject to factors like noisy output, performance, etc.
+
+Some approaches we can take for doing this include:
+
+We then do a depth-first diff of the trees, stopping and reporting on the first different node.
```

Comment:
This feels like it's missing a sentence here?

Reply:
Thanks.  It got moved and lost the context.  Just needed an "After resolving" at the front.